carma_wm_ctrl/src/WMBroadcaster.cpp

```diff
@@ -505,19 +834,33 @@ void WMBroadcaster::setConfigSpeedLimit(double cL)
   config_limit = lanelet::Velocity(cL * lanelet::units::MPH());
 }
 
+uint32_t WMBroadcaster::generate32BitId(const std::string& label)
+{
+  auto pos1 = label.find("INT_ID:");
+  auto pos2 = label.find("SG_ID:");
+  uint16_t intersection_id = std::stoi(label.substr(pos1, pos2 - pos1));
+  uint8_t signal_id = std::stoi(label.substr(pos2, label.size() - pos2));
+
+  uint32_t temp;
+  temp |= intersection_id;
+  temp = temp << 8;
+  temp |= signal_id;
```

Comment:
Ash has this same bit of logic for building temp in his code. Can we add it as a method to carma_wm has a helper method? 
Also please see my comment on lanelet::Id consistency
https://github.com/usdot-fhwa-stol/carma-platform/pull/1355/files#r673302389

Reply:
I refactored the id code out. 
In terms of the id consistency though, I am not sure where to register it. It seems like lanelet::utils::getId() return global variable that is guaranteed to be unique in one node. So I think it wouldn't be necessary to register it, as I am always using that function.
On carma_wm's side too, when the map is update, every primitive goes through `add` function which should be using registerId(). Please let me know if I missed any other place

Reply:
Ok that's fine then. I was under the impression you were using this generated id as the regulatory element id